### ISSUE TITLE
Add support for Catalog in runtime schema mapping

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#4793]
Once we support catalogs in generated SQL (#4792), we must also support mapping them via runtime schema mapping functionality.

This depends on #4795

---

See also: #4792, #4794, #4795

**Comment by luhawk803** [comment]
When can this feature be online?

**Comment by arambau-wgfs** [comment]
That's exactly our use case... and as a paying customer, without catalog mapping, it makes JooQ's generated code not really portable. This should have been integrated long ago. Please prioritize much higher.

**Comment by lukaseder** [comment]
@arambau-wgfs Thank you very much for your message. We understand that this feature means a lot to you and we'll review the priority for an upcoming release.

As a workaround, you can turn off catalog generation in jOOQ's code generator by specifying `<outputCatalogToDefault/>`, see e.g.: https://www.jooq.org/doc/latest/manual/code-generation/codegen-advanced/codegen-config-database/codegen-database-catalog-and-schema-mapping

Alternatively, at runtime, you can turn off the generation of catalog qualification via `Settings.renderCatalog`.

I hope this helps

**Comment by lukaseder** [comment]
As discussed, missing tasks are:

- [ ] SQL Server integration tests similar to `SQLServerTest.testSQLServerMultiDatabase` (we have a table `T_4795` from #4795, which can be reused)
- [ ] Manual documentation will be essential here
- [ ] Possibility to provide both `defaultSchema` and `defaultCatalog`
- [ ] Possibility to provide `MappedCatalog.defaultSchema`
- [ ] Integration test for `MappedCatalog.defaultSchema` to ensure that `[catalog]..[table]` or `[catalog]..[table].[column]` identifiers are possible and work well on SQL Server (I think there's a parser issue for this, too, but cannot seem to find it).

**Comment by lukaseder** [comment]
@knutwannheden I'm preparing 3.13 for the release (it will take a while). Is the status quo in this area releasable with "known issues" that can be backported to 3.13.1? Or should I wait?

**Comment by knutwannheden** [comment]
So far I am not aware of any known issues, but I am currently working on a set of integration tests for SQL Server. For the additional tasks you listed regarding default schema and identifiers like `[catalog]..[table]` I suggest we create a separate issue.

**Comment by lukaseder** [comment]
OK, agreed.

**Comment by lukaseder** [comment]
Closing this already, for the release notes.
